src/components/pages/dashboard/profile.js

```diff
@@ -1,10 +1,202 @@
+    }
+    if (checked === true && data.newPin === "") {
+      data.newPin = registeredUsers[currentUserIndex].accountPin;
+    }
+    if (checked === true && data.newPin === data.confirmNewPin) {
+      registeredUsers[currentUserIndex].accountPin = data.newPin;
+    }
+
+    localStorage.setItem("MB_USER_ACCOUNTS", JSON.stringify(registeredUsers));
+    navigate("/transactions");
+  };
   return (
     <DashboardLayout pageTitle="Profile">
-      <p>This is my profile page</p>
+      <div className="container-fixed">
+        <form className="account-profile" onSubmit={handleSubmit(onSubmit)}>
+          <div className="form-group">
+            <label className="form-control-label">Account Name</label>
+            <input
+              type="text"
+              placeholder={accountName}
+              className="form-control"
+              onChange={handleAccountNameChange}
+              {...register("accountName")}
+            />
+
+            <p className="form-error">{errors.accountName?.message}</p>
+          </div>
+          <div className="form-group">
+            <label className="form-control-label">Account Number</label>
+            <input
+              type="number"
+              id="account-number"
+              className="form-control"
+              value={getLoggedInUserAccountNumber()}
+              readOnly
+            />
+          </div>
+          <label className="form-control-label">
+            <input
+              type="checkbox"
+              className="changePin"
+              onClick={handleToggleChange}
+            />
+            Change Pin
+          </label>
+          {checked && (
+            <div>
+              <div className="form-group">
+                <label className="form-control-label">Current PIN</label>
+                <input
+                  type="password"
+                  maxLength={4}
+                  className="form-control"
+                  onChange={handleCurrentPinChange}
+                  {...register("currentPin")}
+                />
+                {currentPinCode}
+                <p className="form-error">{errors.currentPin?.message}</p>
+              </div>
+              <div className="form-group">
+                <label className="form-control-label">New PIN</label>
+                <input
+                  type="password"
+                  maxLength={4}
+                  className="form-control"
+                  onChange={handleNewPinChange}
+                  {...register("newPin")}
+                />
+                {newPinCode}
+                <p className="form-error">{errors.newPin?.message}</p>
+              </div>
+              <div className="form-group">
+                <label className="form-control-label">Confirm New PIN</label>
+                <input
+                  type="password"
+                  maxLength={4}
+                  className="form-control"
+                  onChange={handleConfirmNewPineChange}
```

Comment:
This is not needed. useForm hook is registering and handling the input changes.

Reply:
Resolved